diff --git a/tools/enigma2.sh.in b/tools/enigma2.sh.in
index 875376e14..3c873a701 100755
--- a/tools/enigma2.sh.in
+++ b/tools/enigma2.sh.in
@@ -41,6 +41,124 @@ while : ; do
 		@bindir@/enigma2_pre_start.sh
 	fi
 
+	if [ -f /etc/resolvconf.conf ]; then
+		if [ ! -f /etc/resolv.conf ]; then
+			ipadress=$( sed -ne 's#.*=\([^"<]*\).*#\1#p' /etc/resolvconf.conf )
+			ln -sfn ${ipadress} /etc/resolv.conf
+		fi
+	fi
+	#  dreambox barry allen mount
+	if [ $(uname -n) == "dm7080" -o $(uname -n) == "dm820" -o $(uname -n) == "dm520" -o $(uname -n) == "dm920" -o $(uname -n) == "dm900" ]; then
+		DISK=$(grep "/media/ba" /proc/mounts | awk '{gsub("/media/ba",""); print $1}' | head -n 1)
+		umount -f /media/ba
+		sleep 1
+		umount -f ${DISK}
+		sleep 1
+		mkdir -p /media/ba
+		sleep 1
+		mount ${DISK} /media/ba
+		sleep 1
+	fi
+	
+	# files Extensions
+	if [ -d "/Extensions" ]                && [[ ! -L "/Extensions" ]]; then
+	    ln -sfn /usr/lib/enigma2/python/Plugins/Extensions /
+	fi
+
+	# python ls files
+	if ! [ -f @bindir@/python ]            && [[ ! -L @bindir@/python ]]; then
+	    if [ -f @bindir@/python3 ]         && [[ -L @bindir@/python3 ]]; then
+                ln -sfn @bindir@/python3 @bindir@/python;
+            fi
+        fi
+	# picon muunt
+	if [ -d "/data/picon" ]                && [[ ! -L "/data/picon" ]]; then
+		picon=/data/picon;
+		ln -sfn ${picon} /usr/share/enigma2/picon;
+		ln -sfn ${picon} /usr/share/enigma2/picon_50x30;
+		ln -sfn ${picon} /usr/share/enigma2/picon_220x132;
+	elif [ -d "/picons/piconSList" ]       && [[ ! -L "/picons/piconSList" ]] ; then
+		picon=/picons/piconSList;
+		ln -sfn ${picon} /usr/share/enigma2/picon;
+		ln -sfn ${picon} /usr/share/enigma2/picon_50x30;
+		ln -sfn ${picon} /usr/share/enigma2/picon_220x132;
+	elif [ -d "/usr/share/enigma2/picon" ] && [[ ! -L "/usr/share/enigma2/picon" ]]; then
+		picon=/usr/share/enigma2/picon;
+		ln -sfn ${picon} /usr/share/enigma2/picon;
+		ln -sfn ${picon} /usr/share/enigma2/picon_50x30;
+		ln -sfn ${picon} /usr/share/enigma2/picon_220x132;
+	elif [ -d "/media/ba/picon" ]          && [[ ! -L "/media/ba/picon" ]] ; then
+		picon=/media/ba/picon;
+		ln -sfn ${picon} /usr/share/enigma2/picon;
+		ln -sfn ${picon} /usr/share/enigma2/picon_50x30;
+		ln -sfn ${picon} /usr/share/enigma2/picon_220x132;
+	elif [ -d "/media/ba/picon_220x132" ]  && [[ ! -L "/media/ba/picon_220x132" ]]; then
+		picon=/media/ba/picon_220x132;
+		ln -sfn ${picon} /usr/share/enigma2/picon;
+		ln -sfn ${picon} /usr/share/enigma2/picon_50x30;
+		ln -sfn ${picon} /usr/share/enigma2/picon_220x132;
+	elif [ -d "/media/hdd/picon" ]         && [[ ! -L "/media/hdd/picon" ]] ; then
+		picon=/media/hdd/picon;
+		ln -sfn ${picon} /usr/share/enigma2/picon;
+		ln -sfn ${picon} /usr/share/enigma2/picon_50x30;
+		ln -sfn ${picon} /usr/share/enigma2/picon_220x132;
+	elif [ -d "/media/hdd/picon_220x132" ] && [[ ! -L "/media/hdd/picon_220x132" ]] ; then
+		picon=/media/hdd/picon_220x132;
+		ln -sfn ${picon} /usr/share/enigma2/picon;
+		ln -sfn ${picon} /usr/share/enigma2/picon_50x30;
+		ln -sfn ${picon} /usr/share/enigma2/picon_220x132;
+	elif [ -d "/media/usb/picon" ]         && [[ ! -L "/media/usb/picon" ]] ; then
+		picon=/media/usb/picon;
+		ln -sfn ${picon} /usr/share/enigma2/picon;
+		ln -sfn ${picon} /usr/share/enigma2/picon_50x30;
+		ln -sfn ${picon} /usr/share/enigma2/picon_220x132;
+	elif [ -d "/media/mmc/picon" ]         && [[ ! -L "/media/mmc/picon" ]] ; then
+		picon=/media/mmc/picon;
+		ln -sfn ${picon} /usr/share/enigma2/picon;
+		ln -sfn ${picon} /usr/share/enigma2/picon_50x30;
+		ln -sfn ${picon} /usr/share/enigma2/picon_220x132;
+	elif [ -d "/media/mmcblk1p1/picon" ]   && [[ ! -L "/media/mmcblk1p1/picon" ]] ; then
+		picon=/media/mmcblk1p1/picon;
+		ln -sfn ${picon} /usr/share/enigma2/picon;
+		ln -sfn ${picon} /usr/share/enigma2/picon_50x30;
+		ln -sfn ${picon} /usr/share/enigma2/picon_220x132;
+	elif [ -d "/media/sdcard/picon" ]      && [[ ! -L "/media/sdcard/picon" ]] ; then
+		picon=/media/sdcard/picon;
+		ln -sfn ${picon} /usr/share/enigma2/picon;
+		ln -sfn ${picon} /usr/share/enigma2/picon_50x30;
+		ln -sfn ${picon} /usr/share/enigma2/picon_220x132;
+	elif [ -d "/picon" ]                   && [[ ! -L "/picon" ]] ; then
+		picon=/picon;
+		ln -sfn ${picon} /usr/share/enigma2/picon;
+		ln -sfn ${picon} /usr/share/enigma2/picon_50x30;
+		ln -sfn ${picon} /usr/share/enigma2/picon_220x132;
+	else
+		rm -rf /usr/share/enigma2/picon > /dev/null 2>&1;
+		picon=/usr/share/enigma2/picon;
+		mkdir -p ${picon} > /dev/null 2>&1;
+		ln -sfn ${picon} /usr/share/enigma2/picon_50x30;
+		ln -sfn ${picon} /usr/share/enigma2/picon_220x132;
+	fi;
+
+	if [ ! $( grep "config.misc.hotkey.green=" /etc/enigma2/settings ) ]; then
+		echo "config.misc.hotkey.green=Module/Screens.PluginBrowser/PluginBrowser" >>/etc/enigma2/settings
+	fi
+	if [ ! $( grep "config.misc.hotkey.yellow=" /etc/enigma2/settings ) ]; then
+		echo "config.misc.hotkey.yellow=Module/Screens.ScanSetup/ScanSetup" >>/etc/enigma2/settings
+	fi
+	if [ ! $( grep "config.usage.frontled_color=" /etc/enigma2/settings ) ]; then
+		echo "config.usage.frontled_color=1" >>/etc/enigma2/settings
+	fi
+	if [ ! $( grep "config.plugins.serviceapp.servicemp3" /etc/enigma2/settings ) ]; then
+		touch /etc/enigma2/serviceapp_replaceservicemp3
+		if [ ! $( grep "config.plugins.serviceapp.servicemp3.player=exteplayer3" /etc/enigma2/settings ) ]; then
+			echo "config.plugins.serviceapp.servicemp3.player=exteplayer3" >>/etc/enigma2/settings
+		fi
+		if [ ! $( grep "config.plugins.serviceapp.servicemp3.replace=true" /etc/enigma2/settings ) ]; then
+			echo "config.plugins.serviceapp.servicemp3.replace=true" >>/etc/enigma2/settings
+		fi
+	fi
+
 	# start enigma
 	sync
 	if [ $DEBUG_LVL -lt 4 ]; then
